### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by andreamlin**
Remove unused methods. Some more lambdas b/c Intellij said so.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java [hunk 1]
```diff
@@ -113,17 +112,11 @@ private boolean isSourceApiInterfaceFile(ProtoFile file) {
...
-      GapicProductConfig productConfig,
-      SurfaceNamer namer,
-      ProtoFile file,
-      boolean hasOverview) {
+      ApiModel model, GapicProductConfig productConfig, SurfaceNamer namer, ProtoFile file) {
     ImmutableList.Builder<ModuleView> moduleViews = ImmutableList.builder();
     for (String moduleName : namer.getApiModules()) {
       if (moduleName.equals(namer.getApiWrapperModuleVersion()) && file != null) {
         moduleViews.add(generateTocModuleView(model, file, productConfig, namer, moduleName));
-      } else if (moduleName.equals(namer.getModuleServiceName()) && hasOverview) {
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: For my own understanding, what was this doing?

**Comment by andreamlin** [comment]
addressed comments
PTAL

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java (hunk 1 above)
Comment (pongad) [reply src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: Hmm yeah you're right. @jbolinger can you confirm that this is safe to do?

**Comment by pongad** [comment]
This LGTM but I think we should have someone familiar with Ruby to sign off.

**Comment by andreamlin** [comment]
@landrito would you mind taking a look at this PR?

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java (hunk 1 above)
Comment (landrito) [reply src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: IIRC this had to do with adding some meta documentation to some generated files. I assume there would have been some feedback saying that it was unnecessary causing the documentation to be removed and this wasn't cleaned up.

**Comment by landrito** [comment]
LGTM assuming that overview view is not being used in templates.

**Pull request merged**